libsolidity/analysis/TypeChecker.cpp

```diff
@@ -629,7 +629,13 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 	}
 	for (ASTPointer<VariableDeclaration> const& var: _function.parameters() + _function.returnParameters())
 	{
-		if (!type(*var)->canLiveOutsideStorage())
+		if (
+			!type(*var)->canLiveOutsideStorage() &&
+			!(
+				(_function.visibility() <= FunctionDefinition::Visibility::Internal) &&
+				type(*var)->category() == Type::Category::Mapping
```

Comment:
We might want to discuss, whether not living outside storage is a problem for *any* type in internal functions.

Reply:
Will create a new issue for removing this check and thinking about ``canLiveOutsideStorage`` in general.

Reply:
Created #4804.